pkg/volume/csi/csi_plugin.go

```diff
@@ -260,11 +274,34 @@ func (p *csiPlugin) NewMounter(
 	spec *volume.Spec,
 	pod *api.Pod,
 	_ volume.VolumeOptions) (volume.Mounter, error) {
-	pvSource, err := getCSISourceFromSpec(spec)
+
+	volSrc, pvSrc, err := getSourceFromSpec(spec)
 	if err != nil {
 		return nil, err
 	}
-	readOnly, err := getReadOnlyFromSpec(spec)
+
+	var (
+		driverName   string
+		volumeHandle string
+		readOnly     bool
+	)
+
+	switch {
+	case volSrc != nil && utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume):
+		volumeHandle = makeVolumeHandle(string(pod.UID), spec.Name())
+		driverName = volSrc.Driver
+		if volSrc.ReadOnly != nil {
+			readOnly = *volSrc.ReadOnly
+		}
+	case pvSrc != nil:
+		driverName = pvSrc.Driver
+		volumeHandle = pvSrc.VolumeHandle
+		readOnly = pvSrc.ReadOnly
```

Comment:
This is the thing that breaks read only tests.  Original `getReadOnlyFromSpec` returned `spec.ReadOnly`, where `spec` is argument of this call, i.e. `spec.VolumeSpec` type (yeah, confusing as hell). It contains ReadOnly flag from `PodSpec`.

Now, `CSIPersistentVolumeSource.ReadOnly` is used and it contains ReadOnly flag from PersistentVolume. Traditionally, we ignore this flag. There was an github issue around that.

Reply:
Which reminds me, should `CSIVolumeSource` include field `ReadOnly` if it's ignored anyway?

Reply:
@jsafrane ignored in what sense? it's passed to the driver which may have a use for it.

Reply:
See https://github.com/kubernetes/kubernetes/issues/70505 for the current confusing read-only situation today.  volumeMount.readonly gets passed to the runtime, but it seems like it relies on the volume plugin to be able to pass that through.... And volumesource.readonly gets passed to the driver, however it doesn't do much for CSI because ro is not mount propagated....

Reply:
> Which reminds me, should CSIVolumeSource include field ReadOnly if it's ignored anyway?

Sorry, I am wrong, inline volumes apparently use that.

Reply:
ah. so we should probably revert the behavior for pv's back to the way it was, and the fix can be dealt with in #70505?